### PR TITLE
TRUNK-4473: Fixed error occuring when user tries to upload binary fil…

### PR DESCRIPTION
**Pull request by ivange94**
…es with form resource

### CODE REVIEW

**Comment by dkayiwa** [comment]
@ivange94 when done with the above, remember to add some unit tests.

**Comment by ivange94** [comment]
Yes, i though the same thing.

**Comment by ivange94** [comment]
@dkayiwa i already had a custom exception on standby, all that is left is the unit test. will make my final commit soon

**Comment by dkayiwa** [comment]
Good thinking! :)

**Comment by dkayiwa** [comment]
Test both the service layer change and the controller.

**Comment by ivange94** [comment]
ok

**Comment by ivange94** [comment]
For the FormServiceImpTest, i know i have to pass an image file to the form resource, so that i can i can simulate what happens when an invalid file is passed to the resource. But the documentation of the org.openmrs.FormResource does not tell which field holds files. Also Since this error is thrown when saving in the database, am i suppose to create a test database?

**Comment by dkayiwa** [comment]
Before committing, did you do an mvn clean install to ensure that the formatting is as per our convention? http://go.openmrs.org/pull-request-tips

File: web/src/test/java/org/openmrs/web/controller/form/FormResourceControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    	byte[] fileContent = "Hello World".getBytes();
+    	String fileName = "/resources/testfile.ico";
+    	MockMultipartHttpServletRequest request = new MockMultipartHttpServletRequest();
+    	MockMultipartFile file = new MockMultipartFile("resourceValue", fileName, "text/plain", fileContent);
+    	
+    	request.setMethod("POST");
+    	request.setParameter("datatype", (new LongFreeTextDatatype()).getClass().getName());
+    	request.setParameter("handler", (new LongFreeTextFileUploadHandler()).getClass().getName());
+    	request.addFile(file);
+
```
Comment (ivange94) [inline web/src/test/java/org/openmrs/web/controller/form/FormResourceControllerTest.java]: @dkayiwa no matter what value i pass to the request and the form resource, can't get it to throw the expected exceptions in test. What am saying is, on the page where the form resource is been saved, if you submit no or invalid file, you get an error immediately due to the exceptions thrown. But when writing the unit tests, no matter what file i pass to the multipart request, no exception is thrown.  Also in the FormServiceImpl.saveFormResource() method, no matter what values i contstruct the resource with, it always seem to complete successfully. I don't know why.

**Pull request merged**